Sources/Misc/Deprecations.swift

```diff
@@ -56,6 +56,14 @@ public extension Purchases {
         return await eligiblePromotionalOffers(forProduct: product)
     }
 
+//    @objc func shouldPurchasePromoProduct(_ product: StoreProduct,
```

Comment:
I'm actually gonna mark this as obsoleted, I don't think it made sense for users to call it anyway, it was more of a programmer error, so better to fail. Breaking compilation in this case I think is warranted.